rust/lancedb/src/remote/table.rs

```diff
@@ -525,11 +525,57 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
             message: "drop_columns is not yet supported.".into(),
         })
     }
+
     async fn list_indices(&self) -> Result<Vec<IndexConfig>> {
-        Err(Error::NotSupported {
-            message: "list_indices is not yet supported.".into(),
-        })
+        // Make request to list the indices
+        let request = self
+            .client
+            .post(&format!("/v1/table/{}/index/list/", self.name));
+        let response = self.client.send(request, true).await?;
+        let response = self.check_table_response(response).await?;
+
+        #[derive(Deserialize)]
+        struct ListIndicesResponse {
+            indices: Vec<IndexConfigResponse>,
```

Comment:
The API actually spells this as "indexes":
```suggestion
            indexes: Vec<IndexConfigResponse>,
```